libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/internal/kapt/KaptWithKotlincTask.kt

```diff
@@ -54,7 +51,10 @@ open class KaptWithKotlincTask : KaptTask(), CompilerArgumentAwareWithInput<K2JV
         val pluginOptionsWithKapt: CompilerPluginOptions = pluginOptions.withWrappedKaptOptions(
             withApClasspath = kaptClasspath,
             changedFiles = changedFiles,
-            compiledSourcesDir = getCompiledSources())
+            classpathChanges = classpathChanges,
+            compiledSourcesDir = getCompiledSources(),
+            processIncrementally = processIncrementally
```

Comment:
Because `KaptTask.isIncremental` keeps value of the flag (`kapt.incremental.apt`) I cannot use it here. `processIncrementally` is evaluated during task execution and indicates that we are indeed able to run incrementally. E.g. right now classpath snapshotting cannot handle added/removed classpath entries, but we can compare current-previous classpath only once we run.

Reply:
Makes sense, thanks for the clarification!